### PR TITLE
[ads] Remove deprecated experiments

### PR DESCRIPTION
**Pull request by aseren**
Uplift request of https://github.com/brave/brave-variations/pull/899

Remove deprecated AdServingStudy because this experiment was only required up until CR109. Ads no longer allows token refills on CR112 or older.

Remove deprecated AdNotificationsStudy because this study was never used due to removing a/b testing.

Remove deprecated EpsilonGreedyBanditStudy because this feature is no longer in use and disabled by default client-side.

### CODE REVIEW

**Review by tmancey — APPROVED** [review]
LGTM

**Review by kjozwiak — APPROVED** [review]
Uplift into `production` approved after deliberating with @brave/uplift-approvers. QA also double checked/ran through a quick sanity check via https://github.com/brave/brave-variations/pull/899#issuecomment-1932832015.

**Pull request merged**
